client/lib/wpcom-undocumented/lib/undocumented.js

```diff
@@ -80,27 +80,6 @@ Undocumented.prototype.settings = function ( siteId, method = 'get', data = {},
 	return this.wpcom.req.post( { path }, { apiVersion }, body, fn );
 };
 
-/**
- * Determine whether a domain name is available for registration
- *
- * @param {string} domain - The domain name to check.
- * @param {number} blogId - Optional blogId to determine if domain is used on another site.
- * @param {boolean} isCartPreCheck - specifies whether this availability check is for a domain about to be added to the cart.
- * @param {Function} fn The callback function
- * @returns {Promise} A promise that resolves when the request completes
- */
-Undocumented.prototype.isDomainAvailable = function ( domain, blogId, isCartPreCheck, fn ) {
```

Comment:
In #56379, a `isDomainAvailable` method was introduced to `wpcom.domain()`. It seems to be a clone of this one. Have you considered also refactoring that one as part of this PR? It's a good opportunity IMHO.

Reply:
I can do that. Just to clarify, are you suggesting another `.then()`/`.catch()` refactor here? Or doing a migration out of `wpcom.domain()` similar to what we're doing with `wpcom.undocumented()`?

Reply:
> Or doing a migration out of wpcom.domain() similar to what we're doing with wpcom.undocumented()?

Precisely this.

Reply:
With some guidance (p1637769031042500-slack-C0761SX4K) from @leonardost and @rafaelgalani (thank you both again!), I've refactored this version of the method as well in ba0b77f8b10538336b82257ff73947e9c1a073b0.

I've also updated the testing instructions in the PR to cover this addition.